src/evo/deterministicmns.cpp

```diff
@@ -1013,7 +1117,10 @@ bool CDeterministicMNManager::IsProTxWithCollateral(const CTransactionRef& tx, u
     if (proTx.collateralOutpoint.n >= tx->vout.size() || proTx.collateralOutpoint.n != n) {
         return false;
     }
-    if (tx->vout[n].nValue != 1000 * COIN) {
+
+    CAmount expectedCollateral = GetMnType(proTx.nType).collat_amount;
```

Comment:
Doesn't seems as expectedCollateral expected to be changed.
`const CAmount expectedCollateral = ...`